ignite/cmd/network_reward_release.go

```diff
@@ -0,0 +1,326 @@
+
+	defaultTestnetGasPrice = "0.0000025stake"
+	defaultSPNGasPrice     = "0.0000025" + networktypes.SPNDenom
+	defaultGasLimit        = 400000
+)
+
+// NewNetworkRewardRelease connects the monitoring modules of launched
+// chains with SPN and distribute rewards with chain Relayer
+func NewNetworkRewardRelease() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "release [launch-id] [chain-rpc]",
+		Short: "Connect the monitoring modules of launched chains with SPN",
+		Args:  cobra.ExactArgs(2),
+		RunE:  networkRewardRelease,
+	}
+
+	c.Flags().AddFlagSet(flagNetworkFrom())
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().String(flagSPNGasPrice, defaultSPNGasPrice, "Gas price used for transactions on SPN")
+	c.Flags().String(flagTestnetGasPrice, defaultTestnetGasPrice, "Gas price used for transactions on testnet chain")
+	c.Flags().Int64(flagSPNGasLimit, defaultGasLimit, "Gas limit used for transactions on SPN")
+	c.Flags().Int64(flagTestnetGasLimit, defaultGasLimit, "Gas limit used for transactions on testnet chain")
+	c.Flags().String(flagTestnetAddressPrefix, cosmosaccount.AccountPrefixCosmos, "Address prefix of the testnet chain")
+	c.Flags().String(flagTestnetAccount, cosmosaccount.DefaultAccount, "testnet chain Account")
+	c.Flags().String(flagTestnetFaucet, "", "Faucet address of the testnet chain")
+	c.Flags().Bool(flagCreateClientOnly, false, "Only create the network client id")
+
+	return c
+}
+
+func networkRewardRelease(cmd *cobra.Command, args []string) (err error) {
+	defer func() {
+		err = handleRelayerAccountErr(err)
+	}()
+
+	session := cliui.New()
+	defer session.Cleanup()
+
+	session.StartSpinner("Setting up chains...")
+
+	launchID, err := network.ParseID(args[0])
+	if err != nil {
+		return err
+	}
+	chainRPC := xurl.HTTPEnsurePort(args[1])
+
+	nb, err := newNetworkBuilder(cmd, CollectEvents(session.EventBus()))
+	if err != nil {
+		return err
+	}
+	n, err := nb.Network()
+	if err != nil {
+		return err
+	}
+	spnChainID, err := n.ChainID(cmd.Context())
+	if err != nil {
+		return err
+	}
+
+	ca, err := cosmosaccount.New(
+		cosmosaccount.WithKeyringBackend(getKeyringBackend(cmd)),
+	)
+	if err != nil {
+		return err
+	}
+
+	if err := ca.EnsureDefaultAccount(); err != nil {
+		return err
+	}
+
+	var (
+		createClientOnly, _ = cmd.Flags().GetBool(flagCreateClientOnly)
+		spnGasPrice, _      = cmd.Flags().GetString(flagSPNGasPrice)
+		testnetGasPrice, _  = cmd.Flags().GetString(flagTestnetGasPrice)
+		spnGasLimit, _      = cmd.Flags().GetInt64(flagSPNGasLimit)
+		testnetGasLimit, _  = cmd.Flags().GetInt64(flagTestnetGasLimit)
+		// TODO fetch from genesis
+		testnetAddressPrefix, _ = cmd.Flags().GetString(flagTestnetAddressPrefix)
+		testnetAccount, _       = cmd.Flags().GetString(flagTestnetAccount)
+		testnetFaucet, _        = cmd.Flags().GetString(flagTestnetFaucet)
+	)
+
+	session.StartSpinner("Creating network relayer client ID...")
+	chain, spn, err := clientCreate(cmd, session, launchID, chainRPC, spnChainID)
```

Comment:
This `clientCreate()` function re-creates the `network.Network`, what about passing the one created just before as a parameter ?

Reply:
The `networkBuilder` created in `clientCreate` is not initialized the same, it takes `CollectEvents(session.EventBus())` as an option, but I'm not sure it's relevant so I will try

Reply:
[f3e3650](https://github.com/ignite/cli/pull/2516/commits/f3e3650fdaa15c019842a2f5ace87b346bcbf47c)